packages/icons/bin/generate-library.js

```diff
@@ -0,0 +1,192 @@
+			'ls-files',
+			'-o',
+			'--full-name',
+			ICON_LIBRARY_DIR,
+		] );
+
+		// Filtering with `grep` in a single `exec` call was tempting, but this
+		// manual filtering avoids any shell escaping weirdness.
+		untrackedFiles = stdout
+			.trim()
+			.split( '\n' )
+			.filter( ( f ) => f.endsWith( '.svg' ) );
+	} catch {
+		return;
+	}
+
+	if ( untrackedFiles.length > 0 ) {
+		throw new Error(
+			`The following SVG files are not under version control:\n\n${ untrackedFiles
+				.map( ( file ) => `  - ${ file }` )
+				.join(
+					'\n'
+				) }\n\nPlease either delete them or add them to Git first:\n\n\tgit add ${ untrackedFiles.join(
+				' '
+			) }\n`
+		);
+	}
+}
+
+async function cleanup() {
+	await execFileAsync( 'git', [ 'clean', '-Xfq', ICON_LIBRARY_DIR ] );
+}
+
+// Generate src/library/*.tsx based on the available SVG files.
+async function generateTsxFiles() {
+	const svgFiles = ( await readdir( ICON_LIBRARY_DIR ) ).filter( ( file ) =>
+		// Stricter than just checking for SVG suffix, thereby avoiding hidden
+		// files and characters that would get in the way of camel-casing.
+		file.match( /^[a-z0-9--]+\.svg$/ )
+	);
+
+	await Promise.all(
+		svgFiles.map( async ( svgFile ) => {
+			const svgPath = path.join( ICON_LIBRARY_DIR, svgFile );
+			const svgContent = await readFile( svgPath, 'utf8' );
+
+			const componentContent = svgToTsx( svgContent );
+			if ( ! componentContent ) {
+				throw new Error(
+					`Could not generate icon element from ${ svgPath }`
+				);
+			}
+
+			const tsxPath = svgPath.replace( /\.svg$/, '.tsx' );
+			await writeFile( tsxPath, componentContent );
+		} )
+	);
+}
+
+// Generate src/library/index.ts as a list of exports of the library's modules.
+async function generateIndex() {
+	const tsxFiles = ( await readdir( ICON_LIBRARY_DIR ) ).filter( ( file ) =>
+		file.endsWith( '.tsx' )
+	);
+
+	let indexTemplate = tsxFiles
+		.map( ( file ) => {
+			const importPath = path.basename( file, '.tsx' );
+
+			// Camel case, but retaining 'RTL' acronym in uppercase
+			const identifier = importPath
+				.replace( /-([0-9A-Za-z])/g, ( _, c ) => c.toUpperCase() )
+				.replace( /Rtl\b/, 'RTL' );
+
+			return `export { default as ${ identifier } } from './${ importPath }';`;
+		} )
+		.join( '\n' );
+
+	// Trailing newlines make ESLint happy
+	indexTemplate += '\n';
+
+	await writeFile( path.join( ICON_LIBRARY_DIR, 'index.ts' ), indexTemplate );
+}
+
+// "Transform" to TSX by interpolating the SVG source into a simple TS module
+// with a single default export.
+//
+// Detect SVG tags like `<circle>` and promote them to WordPress primitives
+// like `<Circle />`, taking care of importing those primitives first.
+function svgToTsx( svgContent ) {
+	let jsxContent = svgContent.trim();
+
+	jsxContent = jsxContent.replace( /\sclass=/g, ' className=' );
+
+	// Tags that ought to be converted to WordPress primitives when converting
+	// SVGs to React elements
+	const primitives = {
```

Comment:
I think now that the mobile gutenberg is not really being developped we should get rid of the primitives package at some point.